HomeTask2.4/cache.js

```diff
@@ -0,0 +1,34 @@
+//https://www.codewars.com/kata/525481903700c1a1ff0000e1/train/javascript
+function cache(func) {
+    const cachedValues = new Map();
+    return (...args) => {
+        //generates the key from arguments
+        const key = args.reduce((ivalue, arg) => {
```

Comment:
Recheck this function as joining numbers directly is not the suggested way. Lets say first calculation is 12 and 2 which equals 14, now you push 122 => 14 in map and next calculation is 1 and 22. In the second case the result is 23, but the map already as 122 which would result in 14(wrong answer). You can update the loop and save as "1,12" as key instead.

Reply:
made change as requested to the fix the key generation by adding "," in between the parameters